vision/vae_mnist/vae_mnist.jl

```diff
@@ -0,0 +1,173 @@
+# Variational Autoencoder(VAE)
+#
+# Auto-Encoding Variational Bayes
+# Diederik P Kingma, Max Welling
+# https://arxiv.org/abs/1312.6114
+
+using Base.Iterators: partition
+using BSON
+using CUDAapi: has_cuda_gpu
+using DrWatson: struct2dict
+using Flux
+using Flux: binarycrossentropy, chunk
+using Flux.Data: DataLoader
+using Images
+using Logging: with_logger
+using MLDatasets
+using Parameters: @with_kw
+using ProgressMeter: Progress, next!
+using TensorBoardLogger: TBLogger, tb_overwrite
+using Random
+
+# load MNIST images and return loader
+function get_data(batch_size)
+    xtrain, ytrain = MLDatasets.MNIST.traindata(Float32)
+    # MLDatasets uses HWCN format, Flux works with WHCN 
+    xtrain = reshape(permutedims(xtrain, (2, 1, 3)), 28^2, :)
```

Comment:
it turns out that MLDatasets already returns in WHCN so no need for `permutedims` here.
Images instead works with HWCN, so the permutedims should go into `convert_to_image`. 
Could you fix this also for the DCGAN and LeNet scripts?

Reply:
Thanks, I removed `permutedims` from `get_data`.